big-o/time-and-space-complexity.md

```diff
@@ -100,7 +100,7 @@ Re-read the examples again, imagining having a million contacts. Did any of thes
 We could consider performance of an algorithm in three different scenarios: best case, worst case and average case scenarios.
 
 - The best case scenario means a scenario where the algorithm finishes the most efficiently it could ever, truly, mathematically be
-- The worst case scenario is the specific scenario where the algorithm would perform the most inefficiently
+- The worst case scenario is the specific scenario where the algorithm would take the maximum amount of time or consume the maximum amount of memory to solve the problem.
```

Comment:
I won't fight for this suggestion, but this bullet point ends up feeling a little long. Would it be easier to read to divide it to something like this?

```suggestion
- The worst case scenario is the specific scenario where the algorithm would perform the most inefficiently in one or both aspects:
  - The algorithm takes the maximum amount of time
  - The algorithm consumes the maximum amount of memory to solve the problem
```

I recognize that my suggestion is... pretty wordy, too. :P Don't use this suggestion if you'd like.